druid/src/widget/svg.rs

```diff
@@ -85,13 +88,29 @@ impl<T: Data> Widget<T> for Svg<T> {
     }
     fn paint(&mut self, paint_ctx: &mut PaintCtx, _data: &T, _env: &Env) {
         //TODO: options for aspect ratio or scaling based on height
-        let scale = paint_ctx.size().width / self.get_size().width;
+        let scalex = paint_ctx.size().width / self.get_size().width;
+        let scaley = paint_ctx.size().height / self.get_size().height;
+        let scale = scalex.min(scaley);
 
         let origin_x = (paint_ctx.size().width - (self.get_size().width * scale)) / 2.0;
         let origin_y = (paint_ctx.size().height - (self.get_size().height * scale)) / 2.0;
         let origin = Point::new(origin_x, origin_y);
 
-        self.svg_data.to_piet(scale, origin, paint_ctx);
+        let clip_rect = RoundedRect::from_origin_size(
+            Point::ORIGIN,
+            Size::new(paint_ctx.size().width, paint_ctx.size().height).to_vec2(),
+            2.,
+        );
+
+        // The SvgData's to_piet function dose not clip to the svg's size
+        // CairoRenderContext is very like druids but with some extra goodies like clip
+        paint_ctx
```

Comment:
You shouldn't need to import piet-cairo, the whole point of piet is to abstract over backends like this. If you use a backend explicitly then druid becomes non-portable.

In this case I believe `paint_ctx.clip(clip_rect);` is what you want.

Reply:
Ah brill I'll see if i can fix this up